### PR TITLE
Windows: Only terminate display for last instance

### PR DESCRIPTION
**Pull request by knopp**
Fixes https://github.com/flutter/flutter/issues/79955

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

**Comment by clarkezone** [comment]
thx.  Will take a look after work this evening

File: shell/platform/windows/angle_surface_manager.cc [hunk 1]
```diff
@@ -200,7 +204,11 @@ void AngleSurfaceManager::CleanUp() {
   }
 
   if (egl_display_ != EGL_NO_DISPLAY) {
-    eglTerminate(egl_display_);
+    // Display is reused between instances so only terminate display
+    // if destroying last instance
+    if (instance_count_ == 1) {
```
Comment (clarkezone) [inline shell/platform/windows/angle_surface_manager.cc:209]: as defense in depth, worth setting count to 0 when you detect this?

**Review by clarkezone — APPROVED** [review]
LGTM with one minor question.

File: shell/platform/windows/angle_surface_manager.cc (hunk 1 above)
Comment (knopp) [reply shell/platform/windows/angle_surface_manager.cc:209]: Not sure I see the point. It is decremented to zero in destructor right after CleanUp returns. This all happens on platform thread so there shouldn't be any concurrency issues.

**Pull request merged**